adi/ad4630.py

```diff
@@ -25,25 +26,27 @@ class ad4630(rx, context_manager, attribute):
 
     """ AD4630 is low power 24-bit precision SAR ADC """
 
+    _compatible_parts = ["ad4630-24", "ad4030-24", "ad4630-16"]
     _complex_data = False
     _data_type = np.uint32
     _device_name = ""
     _rx_channel_names = []
 
     """ Default part to initialize is ad4630-24. If you don't hardware test fails"""
 
+    def __compat_parts__(self):
+        return ad4630._compatible_parts
+
     def __init__(self, uri="", device_name="ad4630-24"):
 
         context_manager.__init__(self, uri, self._device_name)
 
-        compatible_parts = ["ad4630-24", "ad4030-24", "ad4630-16"]
-
-        if device_name not in compatible_parts:
+        if device_name not in self.__compat_parts__(self):
             raise Exception(
                 "Not a compatible device: "
                 + str(device_name)
                 + ". Please select from "
-                + str(compatible_parts)
+                + str(self.__compat_parts__(self))
```

Comment:
Just use the class property  _compatible_parts directly here. There is no point making an extra method here.

Reply:
If I replace `self.__compat_parts__(self)` by `_compatible_parts` then `pre-commit run --all-files` says `_compatible_parts` is undefined. I could try `ad4630._compatible_parts` but then I presume the call from the child class would go wrong.